com.unity.render-pipelines.high-definition/Editor/AssetProcessors/MaterialPostProcessor.cs

```diff
@@ -185,6 +422,26 @@ static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAsse
                     wasUpgraded = true;
                 }
 
+                if (isMaterialUsingPlugin)
+                {
+                    int hdPluginMaterialVersion = (int)(PluginMaterial.GenericVersions.NeverMigrated);
+                    bool neverMigrated = (assetVersion.hdPluginSubTargetMaterialVersions.Count == 0)
+                        || (false == assetVersion.hdPluginSubTargetMaterialVersions.TryGetValue(subTargetGUID, out hdPluginMaterialVersion));
+                    if (neverMigrated)
+                    {
+                        assetVersion.hdPluginSubTargetMaterialVersions.Add(subTargetGUID, hdPluginMaterialVersion);
+                    }
+
+                    if (hdPluginMaterialVersion < subTargetMaterialUtils.latestMaterialVersion)
+                    {
```

Comment:
Below is the main forward hook where the subtarget will process whatever migration on its *material* it needs to do, based on the last seen version (for its guid) that was recorded in the AssetVersion that is stashed inside the material.